cards/tests/dominion/test_cellar.py

```diff
@@ -0,0 +1,63 @@
+from unittest import mock
+
+from cards.kingdom_cards.dominion import Cellar
+from games.factories import GameFactory
+from players.factories import PlayerFactory
+from testing import BaseTestCase
+from turns.factories import AdHocTurnFactory, TurnFactory
+
+
+class CellarCardTestCase(BaseTestCase):
+    def setUp(self):
+        super().setUp()
+        self.player = PlayerFactory()
+        self.game = GameFactory(players=[self.player] + PlayerFactory.create_batch(11))
+        self.turn = TurnFactory(player=self.player, game=self.game)
+        self.deck = self.game.decks.get(player=self.player)
+        self.card = Cellar()
+
+    def test_perform_specific_action(self):
+        with self.assert_adhoc_turn_created():
+            adhoc_turn = self.card.perform_specific_action(
+                deck=self.deck,
+                turn=self.turn,
+            )
+        self.assert_adhoc_turn(
+            adhoc_turn=adhoc_turn,
+            turn=self.turn,
+            player=self.player,
+            game=self.game,
+            card=self.card,
+        )
+
+
+class CellarFormTestCase(BaseTestCase):
+    def setUp(self):
+        super().setUp()
+        self.player = PlayerFactory()
+        self.game = GameFactory(players=[self.player] + PlayerFactory.create_batch(11))
+        self.turn = TurnFactory(player=self.player, game=self.game)
+        self.deck = self.game.decks.get(player=self.player)
+        self.card = Cellar()
+        self.adhoc_turn = AdHocTurnFactory(
+            player=self.player,
+            game=self.game,
+            turn=self.turn,
+            card=self.card,
+        )
+
+    def test_cards_are_discarded(self):
+        form = self.build_card_form(adhoc_turn=self.adhoc_turn, cards=['Copper'])
+        assert form.is_valid()
+        with mock.patch(
+            'decks.models.Deck.discard_cards',
+        ) as discard_cards, mock.patch(
+            'decks.models.Deck.draw_cards',
+        ) as draw_cards:
+            form.save()
+        discard_cards.assert_called_once_with(mock.ANY)
```

Comment:
If we don't care what argument is being passed in, I believe we can use `assert_called_once`.